### ISSUE TITLE
steroids.logger.log sometimes loses log events 

### ISSUE THREAD

**Issue by pre** [AppGyver/steroids#765]

**Comment by Leeds-eBooks** [comment]
Seems to be that this happens when called from the `.then()` method of a Promise. Could this be feasible? I could easily be wrong though, and I don't think it happens all the time even in this circumstance. I'll report back if I find out more.

**Comment by pre** [comment]
From [forums](https://forums.appgyver.com/#!/supersonic:supersonicloggerlog-not-w): 

It seems that if you abstract the call to `logger.log` and wrap it over a `setTimeout` then the log messages will always arrive. Ie.

```
setTimeout(function(){
  	supersonic.logger.log('y hellow' );
},1);
```

**Comment by terwey** [comment]
I'm also experiencing this issue. I am unable to log at all unless using the ```setTimeout()``` trick.

**Comment by battmanz** [comment]
I tried the setTimeout trick and sometimes it works, sometimes it doesn't. Even when it does work, I do not see the log on the connect screen. I can only see it if I open up the steroids.log file. I'm on Windows 8, by the way.

**Comment by Leeds-eBooks** [comment]
Yep same here. I've pretty much stopped using it altogether, which is unfortunate as alerts are less than ideal!

**Comment by pre** [comment]
Just to clarify: Can you see the missing lines in `steroids.log`? Ie. are the log lines missing only from the Connect Screen (in browser) or also from the log file?

As a workaround, you can use `tail -f log/steroids.log` -- afaik, the log events are only missing in the browser. Please report here if that's not true.

**Comment by RPieter** [comment]
From what I can see, they're not in the steroids.log-file.

My testcase: 
`
angular.module('POC').controller("TestController", function ($scope, $http, supersonic) {
	$scope.detectedPlatform = 'detecting...';
	$scope.start = function () {
		supersonic.logger.debug('start license check');
		supersonic.device.platform().then(function (platform) {
			supersonic.logger.debug(
				"Name: " + platform.name + "\n" +
				"Version: " + platform.version + "\n" +
				"Model: " + platform.model
			);
			$scope.detectedPlatform = platform.name;
		});
	}
});`
The first debug-line `start license check` is NOT logged (not in file, nor in connect-screen), the second line (device-info) IS shown.

So it looks like a timing-issue.
However, the setTimeout workaround doesn't help in this case...
